src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -416,6 +416,31 @@ public static IntPtr AllocateTypeAssociatedMemory(Type type, int size)
                 }
             }
         }
+
+        public static unsafe object? Box(ref byte target, RuntimeTypeHandle type)
+        {
+            if (type.IsNullHandle())
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            RuntimeType rtType = (RuntimeType)Type.GetTypeFromHandle(type)!;
```

Comment:
The argument checks should work on `TypeHandle` and `MethodTable*`. Going to through RuntimeType for each of the error handling checks is unnecessary overhead.